### PR TITLE
Pull config and kafka writer out of context

### PR DESCRIPTION
**Pull request by nathan-artie**
Also moves postgres.Run() to sources/postgres

### CODE REVIEW

File: sources/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -23,13 +24,15 @@ type Store struct {
...
-	tableName string
-	streamArn string
-	batchSize uint
-	streams   *dynamodbstreams.DynamoDBStreams
-	storage   *offsets.OffsetStorage
-	shardChan chan *dynamodbstreams.Shard
+	tableName   string
+	streamArn   string
+	batchSize   uint
+	topicPrefix string
```
Comment (nathan-artie) [inline sources/dynamodb/dynamodb.go:32]: Added `topicPrefix`.

File: lib/kafkalib/writer.go [hunk 2]
```diff
@@ -26,12 +26,8 @@ type BatchWriter struct {
...
-	w, err := NewWriter(ctx, cfg)
-	if err != nil {
-		return BatchWriter{}, err
-	}
-	return BatchWriter{w, ctx, cfg}, nil
+func NewBatchWriter(ctx context.Context, cfg config.Kafka, writer *kafka.Writer) BatchWriter {
+	return BatchWriter{writer, ctx, cfg}
 }
 
 func (w *BatchWriter) reload() error {
```
Comment (Tang8330) [inline lib/kafkalib/writer.go:33]: Unrelated to this PR - but can we be consistent about receiver names? Some using `w`, `k`, etc for `*BatchWriter`

File: main.go [hunk 3]
```diff
@@ -32,25 +60,23 @@ func main() {
...
 
-	var statsD *mtr.Client
-	if cfg.Metrics != nil {
-		slog.Info("Injecting datadog")
-		_statsD, err := mtr.New(cfg.Metrics.Namespace, cfg.Metrics.Tags, 0.5)
-		if err != nil {
-			logger.Fatal("Failed to create datadog client", slog.Any("err", err))
-		}
-		statsD = &_statsD
+	kafka, err := setUpKafka(ctx, cfg.Kafka)
```
Comment (Tang8330) [inline main.go:70]: `_kafka`? so it doesn't collide with the import

File: sources/dynamodb/dynamodb.go (hunk 1 above)
Comment (Tang8330) [reply sources/dynamodb/dynamodb.go:32]: Now that we're injecting a bunch of variables inside of *Store, can we add a Validate function? So we ensure things like topicPrefix can never be empty str

**Review by Tang8330 — APPROVED** [review]
lgtm % c

File: lib/kafkalib/writer.go (hunk 2 above)
Comment (nathan-artie) [reply lib/kafkalib/writer.go:33]: Good catch!

File: sources/dynamodb/dynamodb.go (hunk 1 above)
Comment (nathan-artie) [reply sources/dynamodb/dynamodb.go:32]: 👍 I'll add one in a new PR.

**Pull request merged**